cli/cmd/encore/cmdutil/cmdutil.go

```diff
@@ -16,41 +18,54 @@ import (
 	"encr.dev/pkg/errlist"
 )
 
-// AppRoot determines the app root by looking for the "encore.app" file,
+var (
+	ErrNoEncoreApp    = errors.New("no encore.app found in directory (or any of the parent directories)")
+	ErrEncoreAppIsDir = errors.New("encore.app is a directory, not a file")
+)
+
+// MaybeAppRoot determines the app root by looking for the "encore.app" file,
 // initially in the current directory and then recursively in parent directories
 // up to the filesystem root.
 //
 // It reports the absolute path to the app root, and the
 // relative path from the app root to the working directory.
-//
-// On errors it prints an error message and exits.
-func AppRoot() (appRoot, relPath string) {
+func MaybeAppRoot() (appRoot, relPath string, err error) {
```

Comment:
I know I approved this in the other PR, but I wonder if we should look for a `.git` directory in this directory or the directory above?

Then when we create the app on the server, we can automatically set the "folder" within the repo to match the folder we've found, we could also automatically add `encore` as a push target.

Finally if we detect the existing app origin is github, we could display a link to the "Link your Github repo to get free PR environments"